src/GlobalStateProvider.tsx

```diff
@@ -70,16 +62,46 @@ const GlobalStateProvider = ({
     }
   }, [state?.tenantName]);
 
+  const {
+    state: account,
+    set,
+    clear,
+  } = useLocalStorage<string | undefined>({
+    key: `${storageKeys.ACCOUNT}-${state.tenantName}`,
+  });
+
+  useEffect(() => {
+    const run = async () => {
+      if (!account) return;
+      const name = storageService.get('@talisman-connect/selected-wallet-name');
+      if (!name) return;
+      const wallet = getWalletBySource(name);
+      if (!wallet) return;
+      // TODO: optimize this
```

Comment:
How? Or what exactly do you want to optimize?

Reply:
Problem here is that is beinging used in multiple parts of the app and just want to create something reusable and standardized.

Reply:
Regarding yarn we can in package json define what version should be used and that will be enforced than. What version are you guys using and lets agree on that.

Reply:
Is there a good reason to stick to yarn v1 lock files? 
My `yarn --version` says `4.0.0-rc.38`

Reply:
Not really. I am working eith some projects that use v1 as its still more wide spread. Bilut yeah we can upgrade here